app/src/audio/ogg_player.rs

```diff
@@ -0,0 +1,52 @@
+use crate::raw_data_store::AssetName;
+use alloc::sync::Arc;
+use libc::c_void;
+use ogc_rs::prelude::Asnd;
+use ogglib::*;
+
+/**
+ * OGG audio file player. Interfaces with the ogglib doing the actual work.
+ */
+pub struct OGGPlayer {
+    _asnd: Arc<Asnd>,
+}
+
+impl OGGPlayer {
+    /**
+     * Accept the asnd, as our library will use it and only once can own it at a time.
+     */
+    pub fn new(asnd: Asnd) -> Self {
+        Self {
+            _asnd: Arc::<Asnd>::new(asnd),
+        }
+    }
+
+    /**
+     * Play the given OGG audio file. Set the audio to looping to play it infinitely.
+     */
+    pub fn play(&self, audio: &AssetName, is_looping: bool) {
```

Comment:
Nitpick: Instead of a boolean to indiciate looping or not, a dedicated Enum 

```rust
pub enum LoopingMode{
  pub Infinite,
  pub OneTime,
}
```

might be nicer.

Obviously non-blocking 😊

Reply:
Good suggestion, nicer than the magic bool at the end, applied it :)